service/src/main/java/com/ge/predix/acs/attribute/connector/management/AttributeConnectorServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+                    "Unable to persist connector configuration for resource attributes for zone '%s'",
+                    zoneEntity.getName());
+            throw new AttributeConnectorException(message, e);
+        }
+        return isCreated;
+    }
+
+    @Override
+    public AttributeConnector retrieveResourceConnector() {
+        ZoneEntity zoneEntity = this.zoneResolver.getZoneEntityOrFail();
+        return this.connectorConverter.toConnector(zoneEntity.getResourceAttributeConnector());
+    }
+
+    @Override
+    public Boolean deleteResourceConnector() {
+        ZoneEntity zoneEntity = this.zoneResolver.getZoneEntityOrFail();
+        boolean isDeleted = false;
+        if (null == zoneEntity.getResourceAttributeConnector()) {
+            return isDeleted;
+        }
+        zoneEntity.setResourceAttributeConnector(null);
+        try {
+            this.zoneRepository.save(zoneEntity);
+            isDeleted = true;
+        } catch (Exception e) {
+            String message = String.format(
+                    "Unable to delete connector configuration for resource attributes for zone '%s'",
+                    zoneEntity.getName());
+            throw new AttributeConnectorException(message, e);
+        }
+        return isDeleted;
+
+    }
+
+    private void validateAdapterEntityOrFail(final AttributeAdapterConnection adapter) {
+        if (adapter == null) {
+            throw new AttributeConnectorException("Attribute connector configuration requires at least one adapter");
+        }
+        if (adapter.getAdapterEndpoint() == null || adapter.getAdapterEndpoint().isEmpty()) {
+            throw new AttributeConnectorException("Attribute adapter configuration requires a nonempty endpoint URL");
+        }
+        if (adapter.getUaaTokenUrl() == null || adapter.getUaaTokenUrl().isEmpty()) {
+            throw new AttributeConnectorException("Attribute adapter configuration requires a nonempty UAA token URL");
+        }
+        if (adapter.getUaaClientId() == null || adapter.getUaaClientId().isEmpty()) {
+            throw new AttributeConnectorException("Attribute adapter configuration requires a nonempty client ID");
+        }
+        if (adapter.getUaaClientSecret() == null || adapter.getUaaClientSecret().isEmpty()) {
+            throw new AttributeConnectorException("Attribute adapter configuration requires a nonempty client secret");
+        }
+    }
+
+    private void validateConnectorConfigOrFail(final AttributeConnector connector) {
+        if (connector == null) {
+            throw new AttributeConnectorException("Attribute connector configuration requires a valid connector");
+        }
+        if (connector.getAdapters() == null || connector.getAdapters().isEmpty()) {
+            throw new AttributeConnectorException("Attribute connector configuration requires at least one adapter");
+        }
+        if (connector.getMaxCachedIntervalMinutes() < CACHED_INTERVAL_THRESHOLD_MINUTES) {
```

Comment:
lets use jackson for all this validation. Also, if the value of maxCachedIntervalMinutes is not set - it should default to 480 minutes (8hrs).  I propose a minimum of 60 mins ?